### PR TITLE
Simple finetuning test

### LINKED ISSUE #37: Simple test of whether it is finetuning
# Describe the Feature
I would like a python file that takes in the original model and the fine-tuned model after finetuning and assess whether they are significantly different. When I was working on #5 and #16, I kept running into the "is this actually finetuning" problem. Some ideas from Claude are here: https://claude.ai/share/fe727928-a8d5-412b-990b-5a9a6b8a44c4 (maybe perplexity?) 

# Acceptance Criteria
1. It is easy to run and we can run this before evaluation using the path of the original model and the fine tuned model, and it gives us a warning if there are any red flags/throws an error

### PR DESCRIPTION
**Pull request by alessandramaranca**
Closes #37 

# Description

Added a script and a readme file detailing the simple finetuning test idea outlined in #37.  This script compares a base language model and its fine-tuned adapter version using a user-specified input prompt. It calculates perplexity for both models to assess differences in next-token prediction confidence and computes the mean squared error (MSE) between their logits to measure internal behavioral divergence. Low MSE and near-identical perplexity indicate that the adapter likely failed to change the model, while significant differences suggest successful fine-tuning. These two metrics together provide an interpretable diagnostic for whether fine-tuning had any real effect on the model for a prompt given by the user.

## New Dependencies

None

# Testing Instructions

Included in the readme file.

### CODE REVIEW

File: tests/finetuning/is_this_a_finetuned_model.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    adapter_ppl = compute_perplexity(adapter_logits, input_ids)
+    mse_diff = F.mse_loss(base_logits, adapter_logits).item()
+
+    print("\n=== Model Comparison ===")
+    print(f"Input: {prompt}")
+    print(f"Base Model Perplexity:    {base_ppl:.3f}")
+    print(f"Adapter Model Perplexity: {adapter_ppl:.3f}")
+    print(f"Logit MSE Difference:     {mse_diff:.6f}")
+
+    if abs(base_ppl - adapter_ppl) < 0.01 and mse_diff < 1e-4:
```
[the remark this answers is not in the document]
Comment (niznik-dev) [reply tests/finetuning/is_this_a_finetuned_model.py:57]: Even if they don't change it'll probably make the code more readable

[the remark this answers is not in the document]
Comment (alessandramaranca) [reply tests/finetuning/is_this_a_finetuned_model.py]: Made the change and added the argument in the read me and in the base file
